### DISCUSSION TITLE
tis-the-season/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#29] [Announcements]
# tis-the-season/

Remember Way Back When…

https://www.supergoodcode.com/tis-the-season/

**Comment by solerman** [comment]
Can You please get rid of the max 740px width in your blog? It's a pain to read the stats with horizontal scrolling

**Reply by zmike** [reply]
I didn't realize it was that bad. I bumped it up a bit.

**Reply by solerman** [reply]
It's better, but still scrolling. Maybe use a % of screen size or something so it actually uses the width in a maximed window

**Reply by qwertychouskie** [reply]
The real issue is that the tables have a ton of unused empty space in the middle.  @zmike maybe vkoverhead's output can be tweaked so it outputs a valid Markdown table?

Something like this:
```
|    |                          |          |        |
|----|--------------------------|----------|--------|
| 40 | submit_noop              | 19569683 | 100.0% |
| 41 | submit_50noop            | 402324   | 2.1%   |
| 42 | submit_1cmdbuf           | 51356    | 0.3%   |
| 43 | submit_50cmdbuf          | 1840     | 0.0%   |
| 44 | submit_50cmdbuf_50submit | 1031     | 0.0%   |
| ↓  |                          |          |        |
| 40 | submit_noop              | 21008648 | 100.0% |
| 41 | submit_50noop            | 4866415  | 23.2%  |
| 42 | submit_1cmdbuf           | 51294    | 0.2%   |
| 43 | submit_50cmdbuf          | 1823     | 0.0%   |
| 44 | submit_50cmdbuf_50submit | 1828     | 0.0%   |
```
Which outputs this:

|    |                          |          |        |
|----|--------------------------|----------|--------|
| 40 | submit_noop              | 19569683 | 100.0% |
| 41 | submit_50noop            | 402324   | 2.1%   |
| 42 | submit_1cmdbuf           | 51356    | 0.3%   |
| 43 | submit_50cmdbuf          | 1840     | 0.0%   |
| 44 | submit_50cmdbuf_50submit | 1031     | 0.0%   |
| ↓  |                          |          |        |
| 40 | submit_noop              | 21008648 | 100.0% |
| 41 | submit_50noop            | 4866415  | 23.2%  |
| 42 | submit_1cmdbuf           | 51294    | 0.2%   |
| 43 | submit_50cmdbuf          | 1823     | 0.0%   |
| 44 | submit_50cmdbuf_50submit | 1828     | 0.0%   |

**Reply by qwertychouskie** [reply]
https://github.com/zmike/vkoverhead/pull/24

ur welcome, now vkoverhead is truly ready to defeat big triangle

**Comment by jams3223** [comment]
Mike be flexing on us nowadays.

**Comment by HenrikDK2** [comment]
Great work, I love reading your optimization blog posts.

**Comment by roadapathy** [comment]
I'm liking this. I like it a lot. When would we see these improvements in MESA?

**Reply by stevenhoving** [reply]
🚀

**Comment by MadCatX** [comment]
Am I missing something or did the #43 test get slower in every test case? Or is this not a concern?

**Reply by zmike** [reply]
The only cases where it went down by a measurable amount (as opposed to noise) were lavapipe and turnip, and these seems to only occur when the driver is memory bottlenecked, i.e., not real world cases. I'm sure it'd be possible to mitigate this, but I don't know that it's worth the time considering these numbers are only achievable for queue submissions which have zero commands recording in between.

**Comment by hardBSDk** [comment]
How much FPS it could increase in the ANV driver?

**Comment by qwertychouskie** [comment]
"Remember Way Back When…" this blog post was made?  Crazy to think it was over 2 weeks ago...  Where's a dude to find fresh "haha optimizations go brrrrr" content?

;)

**Reply by zmike** [reply]
I know, I know, I'm hoping to get a post out today but it's hard when I'm buried under all this P E R F
